### ISSUE TITLE
Missing some IP addresses, which are needed when using Connect to sync between O365 and AAD

### ISSUE THREAD

**Issue by susan-wangyi** [MicrosoftDocs/OfficeDocs-Enterprise#746]
My customer needs to use Connect to sync between O365 and AAD, and follow this docs to allow mentioned IP ranges to access in the firewall. But still got connection failed error as Connect was unable to connect login.chinacloudapi.cn and adminwebservice.partner.microsoftonline.cn, and the related IP addresses should be:
139.217.231.208
         139.217.228.95
          139.217.231.198
          139.217.231.219
          139.217.118.54
          139.217.118.46
          139.217.118.25
          139.217.115.121

Could you please help double check whether other IP addresses are missing as well and update all needed IP addressed in this doc? Thanks in advance.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0dd0d24e-7a31-8a50-ffdd-7823b6c6c01c
* Version Independent ID: 601aeca3-2c5c-3dda-0402-e4fffaa98ae0
* Content: [URLs and IP address ranges for Office 365 operated by 21Vianet](https://docs.microsoft.com/en-us/office365/enterprise/urls-and-ip-address-ranges-21vianet#feedback)
* Content Source: [Enterprise/urls-and-ip-address-ranges-21vianet.md](https://github.com/MicrosoftDocs/OfficeDocs-Enterprise/blob/live/Enterprise/urls-and-ip-address-ranges-21vianet.md)
* Service: **o365-administration**
* GitHub Login: @LauraWi
* Microsoft Alias: **laurawi**

**Comment by LauraWi** [comment]
@susan-wangyi I'm following up with engineering to investigate. Thanks.

**Comment by LauraWi** [comment]
@susan-wangyi -- Do you have specific information about which app within Office 365 is having the issue? To manage expectations, if there's a true gap in the docs, I can alert the team so they submit the proper endpoints to our publishing tool. If this is blocking customer connectivity, I recommend having the customer work with support.

**Comment by susan-wangyi** [comment]
Hello Laura,
This issue is not specific to an app. Customer needs to use Connect tool to sync local AD with AAD in order to use O365. Customer opened all IPs/ports on their firewall, but find that Connect is trying to connect IPs that I listed before. These IPs are for login.chinacloudapi.cn and adminwebservice.partner.microsoftonline.cn. Connect needs to connect login.chinacloudapi.cn and adminwebservice.partner.microsoftonline.cn for sync. However, these IPs are not listed in this doc. So Could you please help to check and find out whether other IPs are missing as well?

**Comment by LauraWi** [comment]
Tagging in @kelleyvice-msft while Joe is out.

**Comment by kelleyvice-msft** [comment]
Adding @pandrew1 for visibility.
@susan-wangyi, I suspect that the endpoints in Enterprise/urls-and-ip-address-ranges-21vianet.md do not include the IPs you mention because endpoints for AAD are not specific to Office 365. There is a topic that describes how to whitelist AAD endpoints here: https://docs.microsoft.com/en-us/azure/active-directory/hybrid/how-to-connect-install-prerequisites. 
@pandrew1, would you care to elaborate?

**Comment by susan-wangyi** [comment]
@Kelleyvice-msft, yes, I think this is not specific to O365. However, the IPs for AAD are not listed in the doc you mentioned. Also, the URL adminwebservice.partner.microsoftonline.cn is not listed either. My customer firewall needs to know what IPs need to open, rather than URL. So customer wants us to provide a whole list of IPs.

**Comment by susan-wangyi** [comment]
@kelleyvice-msft @pandrew1, may I get some updates here? 
I used nsloopup login.chinacloudapi.cn, and got a list of IPs that I posted before, but all these IPs are not included in this page. 
I will feedback on the page https://docs.microsoft.com/en-us/azure/active-directory/hybrid/how-to-connect-install-prerequisites as well.
Could you please help check all needed IPs and update this page? Really appreciate.

**Comment by susan-wangyi** [comment]
I just searched worldwide endpoints https://docs.microsoft.com/en-us/office365/enterprise/urls-and-ip-address-ranges, which includes adminwebservice.microsoftonline.com.
However, in this page, adminwebservice.partner.microsoftonline.cn is not listed.

**Comment by susan-wangyi** [comment]
@kelleyvice-msft @pandrew1, sorry for posting several comments here, please allow me to summarize my customer requests as below:

Customer firewall needs to know what IPs to open rather than URLs. He followed this page to open all IPs/IP ranges on the firewall, but still failed as Connect unable to connect the IPs:
139.217.231.208
139.217.228.95
139.217.231.198
139.217.231.219
139.217.118.54
139.217.118.46
139.217.118.25
139.217.115.121
These IPs are for login.chinacloudapi.cn and adminwebservice.partner.microsoftonline.cn.

Although this page includes these 2 URLs, it doesn't list the corresponding IPs/IP ranges.
So could you please help check whether other IPs/IP ranges are missing as well, and update this page? Really appreciate.

**Comment by susan-wangyi** [comment]
After comparing the public cloud and Azure China, the following endpoints/IPs are also missing:
mscrl.microsoft.com
*.verisign.com
*.entrust.net
secure.aadcdn.parter.microsoftonline-p.cn

**Comment by kelleyvice-msft** [comment]
@susan-wangyi, I'm evaluating all of this and consulting with the engineering team, who will have the final say on how we address this issue. I will get back to you as soon as I can. Thank you!

**Comment by susan-wangyi** [comment]
@kelleyvice-msft, many thanks for your help.

**Comment by susan-wangyi** [comment]
@kelleyvice-msft, may I get some updates? Thanks in advance.

**Comment by kelleyvice-msft** [comment]
@susan-wangyi, I apologize for the delay. I have been told there is a project underway right now to get these on-prem server to Office 365 service network endpoints published to https://docs.microsoft.com/office365/enterprise/additional-office365-ip-addresses-and-urls in the same way as the endpoints we publish for client machines. In the meantime, I am looking into manually updating the topic with the current endpoints. I don't have any information yet about the timeline for this, but I will find out what I can and respond here when I have more information.

**Comment by susan-wangyi** [comment]
@kelleyvice-msft, thanks a lot for your time and effort on this request.
Could you please first work on the following endpoints as they are the minimum requirements for AAD connect?

mscrl.microsoft.com
*.verisign.com
*.entrust.net
secure.aadcdn.parter.microsoftonline-p.cn
*.chinacloudapi.cn
*.partner.microsoftonline.cn


Since customer firewall can only limit IPs/IP ranges rather than URLs, we need these URLs' corresponding IPs/IP ranges ASAP due to security reasons. Thanks for your understanding.
